### PR TITLE
feat: add extra collateral currencies total value calculation [SF-1194]

### PR DESCRIPTION
**Pull request by leeshuzheng**

### CODE REVIEW

**Review by biga816 — APPROVED** [review]

**Review by biga816 — COMMENTED** [review]
The following code also needs `extraCollateralCurrencies`.

https://github.com/Secured-Finance/secured-finance-app/pull/600/files#diff-e69d70364f2c0acdf79a6612085d7d9d68d0dbdefc7c84d3d7cb6060cc6c573bR162-R163

File: src/components/pages/MarketDashboard/MarketDashboard.tsx [hunk 1]
```diff
@@ -65,8 +66,13 @@ export const MarketDashboard = () => {
     const { data: lendingContracts = baseContracts } = useLendingMarkets();
     const { data: isGlobalItayose } = useIsGlobalItayose();
     const { data: currencies = [] } = useCurrencies();
+    const { data: collateralCurrencies = [] } = useCollateralCurrencies();
     const { data: delistedCurrencySet } = useCurrencyDelistedStatus();
 
+    const extraCollateralCurrencies = collateralCurrencies.filter(
+        element => !currencies.includes(element)
+    );
```
[the remark this answers is not in the document]
Comment (as472) [reply src/components/pages/MarketDashboard/MarketDashboard.tsx:74]: should consider this suggestion from code rabbit. The current implementation is 0(n^2)
Reduce is to O(n)

File: src/components/pages/MarketDashboard/MarketDashboard.tsx [hunk 2]
```diff
@@ -113,7 +119,7 @@ export const MarketDashboard = () => {
         if (!valueLockedByCurrency) {
```
[the remark this answers is not in the document]
Comment (leeshuzheng) [reply src/components/pages/MarketDashboard/MarketDashboard.tsx]: This is done

File: src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    );
+
+    const totalValueLockedInUSD = useMemo(() => {
+        let val = ZERO_BI;
+        if (!valueLockedByCurrency) {
+            return val;
+        }
+        for (const ccy of currenciesInvolved ?? []) {
+            if (!valueLockedByCurrency[ccy]) continue;
+            val = BigInt(
```
Comment (as472) [inline src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.ts:36]: val += and not val =

File: src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { mockUseSF } from 'src/stories/mocks/useSFMock';
+import { renderHook, waitFor } from 'src/test-utils';
+import { ZERO_BI } from 'src/utils';
+import { useTotalValueLockedAndCurrencies } from './useTotalValueLockedAndCurrencies';
+
+const mockSecuredFinance = mockUseSF();
+jest.mock('src/hooks/useSecuredFinance', () => () => mockSecuredFinance);
+describe('useTotalValueLockedAndCurrencies', () => {
```
Comment (as472) [inline src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.test.ts:8]: please use an extra line between 7 and 8

File: src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        });
+    });
+
+    it('should return an array of currencies used for calculating the total value locked', async () => {
+        const { result } = renderHook(() => useTotalValueLockedAndCurrencies());
+        const value = result.current;
+        expect(value.currencies).toEqual([]);
+
+        await waitFor(() => {
+            expect(mockSecuredFinance.getCurrencies).toHaveBeenCalledTimes(2);
```
Comment (as472) [inline src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.test.ts:34]: why twice?

File: src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.test.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            expect(
```
Comment (as472) [inline src/hooks/useTotalValueLockedAndCurrencies/useTotalValueLockedAndCurrencies.test.ts:35]: why twice?

File: src/components/pages/MarketDashboard/MarketDashboard.tsx [hunk 7]
```diff
@@ -66,7 +63,8 @@ export const MarketDashboard = () => {
     const { data: isGlobalItayose } = useIsGlobalItayose();
     const { data: currencies = [] } = useCurrencies();
     const { data: delistedCurrencySet } = useCurrencyDelistedStatus();
-
+    const { totalValueLockedInUSD, currencies: totalValueLockedCurrencies } =
```
Comment (as472) [inline src/components/pages/MarketDashboard/MarketDashboard.tsx:66]: please use this hook in the Global Itayose dashboard also.

**Review by as472 — APPROVED** [review]

**Review by as472 — APPROVED** [review]

**Pull request merged**